packages/flutter/lib/src/widgets/drag_boundary.dart

```diff
@@ -0,0 +1,101 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:ui';
+
+import 'framework.dart';
+
+/// The interface for defining the algorithm that a boundary handles the specified type of shape to be dragged within.
+///
+/// See also:
+///  * [DragBoundary], which uses this class.
+///
+/// `T` is a data class that defines the shape being dragged. For example, when dragging a rectangle within the boundary,
+/// `T` should be a `Rect`.
+abstract class DragBoundaryDelegate<T> {
+  /// Returns whether the specified drag object position is within the boundary.
+  bool isWithinBoundary(T position);
+
+  /// Returns the position of the given drag object after moving it fully inside
```

Comment:
Here we are not returning a position, but the actual dragged object, should we specify that instead? So maybe something like `Returns the given dragged object after moving...`